### PR TITLE
sanitycheck: do not expect results from build_only instances

### PR DESCRIPTION
**Pull request by nashif**
instances that do not run will have no results (beside the fact they
built successfully), so log those as such.

### CODE REVIEW

**Review by galak — APPROVED** [review]

**Review by aasthagr — APPROVED** [review]

File: scripts/sanity_chk/sanitylib.py [hunk 1]
```diff
@@ -2058,6 +2058,7 @@ def process(self, message):
                 self.instance.reason = "Build Failure"
                 pipeline.put({"op": "report", "test": self.instance})
             else:
+                self.instance.status = "passed"
```
Comment (PerMac) [inline scripts/sanity_chk/sanitylib.py:2061]: I wonder if it won't be better if we we only fill the status with 'passed' for instances with build_only == True 
e.g.: 
```
            else:
                if self.instance.build_only:
                    self.instance.status = "passed"
```

Comment (PerMac) [inline scripts/sanity_chk/sanitylib.py:2061]: If this fix is for build_only instances maybe it better to add a specific check for this before filling the status? E.g.:
```
else:
    if self.instance.build_only:
        self.instance.status = "passed"
```

Comment (nashif) [reply scripts/sanity_chk/sanitylib.py:2061]: This status should be set regardless of build_only, we set error above but we set no status when the build is successful.

Comment (nashif) [reply scripts/sanity_chk/sanitylib.py]: addressed

**Review by PerMac — APPROVED** [review]

**Pull request merged**